pkg/controller/operators/catalog/operator.go

```diff
@@ -1056,6 +1067,25 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				// Attempt to create the CRD.
 				_, err = o.OpClient.ApiextensionsV1beta1Interface().ApiextensionsV1beta1().CustomResourceDefinitions().Create(&crd)
 				if k8serrors.IsAlreadyExists(err) {
+					currentCRD, _ := o.OpClient.ApiextensionsV1beta1Interface().ApiextensionsV1beta1().CustomResourceDefinitions().Get(crd.GetName(), metav1.GetOptions{})
+					// Compare 2 CRDs to see if it needs to be updatetd
+					if !reflect.DeepEqual(crd, *currentCRD) {
+						// Verify CRD ownership, only attempt to update if
+						// CRD has only one owner
+						// Example: provided=database.coreos.com/v1alpha1/EtcdCluster
+						apiKey := fmt.Sprintf("provided=%s/%s", crd.APIVersion, crd.Kind)
+						matchedCSV, err := index.APIsIndexValues(o.csvProvidedAPIsIndexer, apiKey)
+						if err != nil {
+							return errorwrap.Wrapf(err, "error find matched CSV: %s", step.Resource.Name)
+						}
+						if len(matchedCSV) == 1 {
+							// Attempt to update CRD
+							_, err = o.OpClient.ApiextensionsV1beta1Interface().ApiextensionsV1beta1().CustomResourceDefinitions().Update(&crd)
```

Comment:
This update will fail because the resource version is empty. You can just copy it from the retrieved version like:
`crd.SetResourceVersion(currentCRD.GetResourceVersion())`.